### PR TITLE
Optimizes calls to Fiat for getAll endpoints that can have large amou…

### PR DESCRIPTION
**Pull request by ttomsu**
…nts of returned data. Cleans up some batchUpdate TODOs.

Also removes a few `deleteById` endpoints because they're not being used as endpoints and would be difficult to authorize.

Metal Travis will choke until Fiat 0.5.0 is released.

@duftler @jtk54 @cfieber PTAL.

### CODE REVIEW

**Comment by ajordens** [comment]
It's not explicitly tied to this PR (_but I was away for the original_) ... but there is a Hystrix-related cache of all applications in `gate` that is used to serve fallbacks amongst other things. 

Not sure how this would play with request-level permissions applied to `list()` in `front50`?

**Comment by jtk54** [comment]
@ajordens @ttomsu sounds like that may also need a Fiat annotation. Think we can annotate just the fallback function rather than duplicating the authZ check for both the Gate and Front50 endpoint.

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy [hunk 1]
```diff
@@ -55,7 +56,7 @@ class StrategyController {
         pipelineStrategyDAO.getPipelinesByApplication(application)
     }
 
-    @PostAuthorize("hasPermission(filterObject.application, 'APPLICATION', 'READ')")
+    @PostFilter("hasPermission(filterObject.application, 'APPLICATION', 'READ')")
```
Comment (jtk54) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy]: Just for my understanding, does `PostFilter` run the function and then do the authZ check? Or the reverse?

**Comment by jtk54** [comment]
LGTM. We should probably address the Hystrix caching business in Gate -- @ajordens is there any other Hystrix caching you're aware of?

**Comment by ajordens** [comment]
There's also the business of `echo` pulling (_and caching_) the super set of all pipelines ... I think there was a plan around that but it escapes me.

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy (hunk 1 above)
Comment (ttomsu) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy]: Yes - it evaluates the SpEL expression passed against the return object.

Comment (jtk54) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/StrategyController.groovy]: Ok makes sense, we need that object's context before we can decide privileges.

**Comment by ttomsu** [comment]
I've opened up two Fiat issues for Echo and Gate's caches. I'll tackle them in due time.

**Pull request merged**